diff --git a/ios/ImageView.swift b/ios/ImageView.swift
index 2ab4cba8309c79e1bff903766e0013d17a3f382e..a93991bc7358cc7f1f5bca2f5f1e9650e36a24db 100644
--- a/ios/ImageView.swift
+++ b/ios/ImageView.swift
@@ -249,7 +249,7 @@ public final class ImageView: ExpoView {
       return nil
     }()
 
-    if let path, let local = UIImage(named: path) {
+    if let path, !path.isEmpty, let local = UIImage(named: path) {
       renderSourceImage(local)
       return true
     }
